test/sanity/code-smell/package-data.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import fnmatch
+import os
+import re
+import tempfile
+import subprocess
+
+
+def main():
+    ignore_files = frozenset((
+        '*/.git_keep',
+        '*/galaxy/data/default/role/*/main.yml.j2',
+        '*/galaxy/data/default/role/*/test.yml.j2',
+        '*/galaxy/data/default/collection/plugins/README.md.j2',
+    ))
+
+    non_py_files = []
+    for root, dirs, files in os.walk('lib/ansible/'):
+        for filename in files:
+            path = os.path.join(root, filename)
+            if os.path.splitext(path)[-1] not in ('.py', '.pyc', '.pyo'):
+                add = True
+                for ignore in ignore_files:
+                    if fnmatch.fnmatch(path, ignore):
+                        add = False
+                if add:
+                    non_py_files.append(path[12:])
```

Comment:
I'd prefer using `os.path` functions for path manipulation instead of string slicing.